aten/src/ATen/native/cuda/MemoryAccess.cuh

```diff
@@ -193,6 +222,54 @@ struct vectorized {
   }
 };
 
+template <int num_outputs, typename data_t, typename dtypes_t, typename offset_calc_t>
+struct multi_outputs_unroll {
```

Comment:
Instead of copy-pasting the implementation, can you modify the original in place to allow multiple outputs? Or another approach could be to make this a subclass of `unroll`.